data/web/diagnostics.php

```diff
@@ -194,25 +194,31 @@
     }
   }
   
-  foreach ($currents as $current) {
-    $current['type'] == strtoupper($current['type']);
+  foreach ($currents as &$current) {
     if ($current['type'] != $record[1])
     {
       continue;
     }
     
-    elseif ($current['type'] == 'TXT' && strpos($record[0], '_dmarc.') === 0) {
+    elseif ($current['type'] == 'TXT' && strpos($current['txt'], 'v=DMARC1') === 0) {
+      $current['txt'] = str_replace(' ', '', $current['txt']);
```

Comment:
I realise this has already been merged, but wouldn't trim() be more appropriate?

Reply:
No, because the spaces can be anywhere in the string. `trim` would only remove them at the beginning and end.

Reply:
I think he was removing the spaces in between the records value.

Reply:
Right, I would have thought spaces in the middle of the record would want to be preserved/checked against?

Additionally line 203 (just above) checks that the record starts with 'v=DMARC1' at pos 0, meaning if there are any leading spaces the record won't be recognised. Perhaps the trimming should happen earlier?

Reply:
Leading spaces are not valid in DMARC, so we don't care about them. The only place where they are optionally possible is after the semicolons. They carry no meaning though, so we can just remove all spaces.

Reply:
Okay, sounds good.